src/analyze/analyze-security.c

```diff
@@ -2380,12 +2382,105 @@ static int get_security_info(Unit *u, ExecContext *c, CGroupContext *g, Security
         return 0;
 }
 
-int analyze_security(sd_bus *bus, char **units, AnalyzeSecurityFlags flags) {
+static int offline_security_check(Unit *u) {
+        _cleanup_(table_unrefp) Table *overview_table = NULL;
+        AnalyzeSecurityFlags flags = 0;
+        _cleanup_(security_info_freep) SecurityInfo *info = NULL;
+        int r = 0, k = 0;
+
+        assert(u);
+
+        if (DEBUG_LOGGING)
+                unit_dump(u, stdout, "\t");
+
+        k = get_security_info(u, unit_get_exec_context(u), unit_get_cgroup_context(u), &info);
+        if (k < 0 && r == 0)
```

Comment:
no need for k here - we don't want to continue if get_security_info fails, as that would be for OOM errors, so just use the usual `r = foo()` pattern here